tools/backup_analysis.py

```diff
@@ -245,69 +246,93 @@ def compare_file(olderdir, newerdir, opts, filename, cmp_size):
+    if not opts.terse:
+        print(f'Total: {total_chg_bytes} bytes changed in {total_chg_blocks} changed granularity-sized ({opts.granularity}) blocks ({chg_blocks}%) of {total_gran_blocks} blocks overall')
+        print(f'Total: {total_files_changed} {plural("file", total_files_changed)} changed out of {total_files} total files')
 
     # Walk through all the types printing out final information per type.
     for n, t in zip(global_names, global_types):
         ts = typestats[t]
         changed = plural('file', ts.files_changed)
         total = plural('file', ts.files)
-        print(f'{n}: {ts.files_changed} {changed} changed out of {ts.files} {total}')
+        if not opts.terse:
+            print(f'{n}: {ts.files_changed} {changed} changed out of {ts.files} {total}')
         if ts.gran_blocks != 0:
             chg_blocks = round(abs(ts.chg_blocks / ts.gran_blocks * 100))
-            print(f'{ts.files_changed} changed {changed}: differs by {ts.chg_blocks} ({chg_blocks}%) granularity blocks in {ts.gran_blocks} total granularity blocks')
-            print(f'{ts.files_changed} changed {changed}: differs by {ts.bytes} bytes in {ts.chg_blocks} changed granularity blocks')
             pct20_blocks = round(abs(ts.pct20 / ts.gran_blocks * 100))
             pct80_blocks = round(abs(ts.pct80 / ts.gran_blocks * 100))
-            print(f'{n}: smallest 20%: {ts.pct20} of {ts.chg_blocks} changed blocks ({pct20_blocks}%) differ by {pct20} bytes or less of {opts.granularity}')
-            print(f'{n}: largest 80%: {ts.pct80} of {ts.chg_blocks} changed blocks ({pct80_blocks}%) differ by {pct80} bytes or more of {opts.granularity}')
+            if not opts.terse:
+                print(f'{ts.files_changed} changed {changed}: differs by {ts.bytes} bytes in {ts.chg_blocks} changed granularity blocks')
+                print(f'{ts.files_changed} changed {changed}: differs by {ts.chg_blocks} ({chg_blocks}%) granularity blocks in {ts.gran_blocks} total granularity blocks')
+                print(f'{n}: smallest 20%: {ts.pct20} of {ts.chg_blocks} changed blocks ({pct20_blocks}%) differ by {pct20} bytes or less of {opts.granularity}')
+                print(f'{n}: largest 80%: {ts.pct80} of {ts.chg_blocks} changed blocks ({pct80_blocks}%) differ by {pct80} bytes or more of {opts.granularity}')
+            else:
+                # Print a terse summary all on one line.
+                all_bytes = opts.granularity * ts.gran_blocks
+                pct_bytes = round(abs(ts.bytes / all_bytes * 100))
+                print(
+                    f'{n}: files: {ts.files_changed} of {ts.files};'
+                    f' bytes: {ts.bytes} of {all_bytes} {pct_bytes}%;'
+                    f' blocks: {ts.chg_blocks} of {ts.gran_blocks} {chg_blocks}%;'
+                    f' {pct20_blocks}% <20%;'
+                    f' {pct80_blocks}% 80>%')
```

Comment:
`80>%` maybe should be `<80%`?

Reply:
I still think it's a bit weird though the +/- makes it less weird.

Here's why I'm confused.  You said:
`what it means is percent of blocks that had less than 20% of the bytes in the block changed`.
Translating the "less than" in that sentence it looks like : `%blocks that had <20%`

whereas: `versus blocks that had more than 80% of the bytes in the block changed`
for me translates as: `%blocks that had >80%`.

Maybe you have a different way to visualize the symbols...?
Ultimately, you're the primary customer of this tool, so whatever makes sense to you is fine with me!

Reply:
I did move the "title" before the percentage to be consistent with the other "titles" in the line FWIW. I went with the "+/-".

Reply:
Yes, that looks better to me!  Thanks.